x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecycleExecutionState.java

```diff
@@ -97,6 +97,15 @@ public static LifecycleExecutionState fromIndexMetadata(IndexMetadata indexMetad
         return fromCustomMetadata(customData);
     }
 
+    /**
+     * Return true if this index is in the frozen phase, false if not controlled by ILM or not in frozen.
+     * @param indexMetadata the metadata of the index to retrieve phase from.
+     * @return true if frozen phase, false otherwise.
+     */
+    public static boolean isFrozenPhase(IndexMetadata indexMetadata) {
+        Map<String, String> customData = indexMetadata.getCustomData(ILM_CUSTOM_METADATA_KEY);
+        return customData != null && TimeseriesLifecycleType.FROZEN_PHASE.equals(customData.get(PHASE));
```

Comment:
Yeah, I was deliberately micro-tuning this a bit, since it will be called rather heavily from the autoscaling decider and I felt like skipping the various object creations and map lookups etc. made sense.

Reply:
Okay, if the performance difference is important enough, we might want to add a comment to make sure it doesn't get code-golfed down to a less optimal version in the future

Reply:
++, added in bfc5fa4bff9b0d0057a265bafbcc764d13e09e3d